src/context.cpp

```diff
@@ -347,11 +364,7 @@ class context_impl_t : public context_t {
 
     execution_unit_t&
     engine() override {
-        typedef std::unique_ptr<execution_unit_t> unit_t;
-        auto comp = [](const unit_t& lhs, const unit_t& rhs) {
-            return lhs->utilization() < rhs->utilization();
-        };
-        return **std::min_element(m_pool.begin(), m_pool.end(), comp);
+        return *m_engine_distributor->next(m_pool);
```

Comment:
What is expected count of execution_unit_t per pools? Would it be any profit of using heap based priority structures?

Reply:
It seems that it could be possibility for m_pool to be empty on incorrect `m_config->network().pool()` parameter, but may be (hopefully) it is some kind of assertion somewhere.

Reply:
```
            m_pool     = source.at("pool", boost::thread::hardware_concurrency() * 2).as_uint();

            if(m_pool <= 0) {
                throw cocaine::error_t("network I/O pool size must be positive");
            }
```